src/components/vm/vmActions.tsx

```diff
@@ -63,140 +66,170 @@ import store from "../../store.js";
+    domainStart({ id: vm.id, connectionName: vm.connectionName }).catch(ex => {
+        setOperationInProgress(false);
+        console.warn("Failed to start VM", vm.name, ":", cockpit.message(ex));
+        store.dispatch(
+            updateVm({
+                connectionName: vm.connectionName,
+                name: vm.name,
+                error: {
+                    text: cockpit.format(_("VM $0 failed to start"), vm.name),
+                    detail: ex.message,
+                }
+            })
+        );
+    })
+);
+
+const onInstall = (vm: VM, onAddErrorNotification: (n: Notification) => void) => (
+    domainInstall({ vm }).catch(ex => {
+        onAddErrorNotification({
+            text: cockpit.format(_("VM $0 failed to get installed"), vm.name),
+            detail: ex.message.split(/Traceback(.+)/)[0],
+            resourceId: vm.id,
         });
+    })
+);
+
+const onReboot = (vm: VM) => (
+    domainReboot({ id: vm.id, connectionName: vm.connectionName }).catch(ex => {
+        store.dispatch(
+            updateVm({
+                connectionName: vm.connectionName,
+                name: vm.name,
+                error: {
+                    text: cockpit.format(_("VM $0 failed to reboot"), vm.name),
+                    detail: ex.message,
+                }
+            })
+        );
+    })
+);
+
+const onForceReboot = (vm: VM) => (
+    domainForceReboot({ id: vm.id, connectionName: vm.connectionName }).catch(ex => {
+        store.dispatch(
+            updateVm({
+                connectionName: vm.connectionName,
+                name: vm.name,
+                error: {
+                    text: cockpit.format(_("VM $0 failed to force reboot"), vm.name),
+                    detail: ex.message,
+                }
+            })
+        );
+    })
+);
+
+const onShutdown = (vm: VM, setOperationInProgress: (val: boolean) => void) => (
+    domainShutdown({ id: vm.id, connectionName: vm.connectionName })
+            .then(() => !vm.persistent && cockpit.location.go(["vms"]))
+            .catch(ex => {
+                setOperationInProgress(false);
+                store.dispatch(
+                    updateVm({
+                        connectionName: vm.connectionName,
+                        name: vm.name,
+                        error: {
+                            text: cockpit.format(_("VM $0 failed to shutdown"), vm.name),
+                            detail: ex.message,
+                        }
+                    })
+                );
+            })
+);
+
+const onPause = (vm: VM) => (
+    domainPause({ id: vm.id, connectionName: vm.connectionName }).catch(ex => {
+        store.dispatch(
+            updateVm({
+                connectionName: vm.connectionName,
+                name: vm.name,
+                error: {
+                    text: cockpit.format(_("VM $0 failed to pause"), vm.name),
+                    detail: ex.message,
```

Comment:
These 8 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2187-561408b0-20250717-145115-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)